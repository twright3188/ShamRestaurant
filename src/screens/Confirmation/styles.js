import { StyleSheet } from 'react-native';
import Colors from '../../theme';

const styles = StyleSheet.create({
  pageStyles: {
    width: '100%',
    height: '100%',
    paddingHorizontal: 15,
    backgroundColor: Colors.confirmationBackgroundColor
  },
  messageStyles: {
    width: '100%',
    backgroundColor: Colors.confirmationMessageBackgroundColor,
    borderRadius: 16,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 5,
    shadowColor: Colors.shadowColor,
    shadowOpacity: 1,
    elevation: 1,
    paddingHorizontal: 16,
    paddingVertical: 24,
    marginBottom: 72,
    marginTop: 80
  },
  textStyles: {
    marginTop: 16
  },
  buttonStyles: {
    height: 56,
  }
});

export default styles;