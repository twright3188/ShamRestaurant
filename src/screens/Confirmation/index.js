import React, { useState, useEffect, useCallback } from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import Title from '../../components/Title';
import Paragraph from '../../components/Paragraph';
import Button from '../../components/Button';
import I18n from 'react-native-i18n';
import styles from './styles';

function Confirmation(props) {
  const { navigation, navigation: { state: { params: { result, email } } } } = props;
  const [payment, setPaymentStatus] = useState(false);  

  const memoizedOrderPayment = useCallback(
    () => props.navigation.navigate('OrderPayment', { result, email }) , 
    [result]
  );

  useEffect(() => {
    if (!payment) {
      setTimeout(() => {
        setPaymentStatus(true);
        memoizedOrderPayment();
      }, 1000);
    }
  }, [payment]);

  return (
    <View style={styles.pageStyles}>
      <View style={styles.messageStyles}>
        <Title>{I18n.t("confirmText1")}</Title>
        <Paragraph style={styles.textStyles}>
            {I18n.t("confirmText2")}
        </Paragraph>
      </View>
       <Button style={styles.buttonStyles} onPress={() => navigation.navigate('Catalog')}>{I18n.t("go_to_catalogue")}</Button>
    </View>
  );
};

Confirmation.propTypes = {
  navigation: PropTypes.instanceOf(Object).isRequired,
};

export default Confirmation;