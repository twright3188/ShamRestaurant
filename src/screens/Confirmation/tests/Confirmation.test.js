import React from 'react';
import renderer from 'react-test-renderer';
import Confirmation from '../index';

it('confirmation screen renders correctly', () => {
    const tree = renderer.create(<Confirmation />).toJSON();
    expect(tree).toMatchSnapshot(); 
});

