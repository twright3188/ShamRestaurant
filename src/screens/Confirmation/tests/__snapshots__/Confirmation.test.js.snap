// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`confirmation screen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F2F7F4",
      "height": "100%",
      "paddingHorizontal": 15,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 16,
        "elevation": 1,
        "marginBottom": 72,
        "marginTop": 80,
        "paddingHorizontal": 16,
        "paddingVertical": 24,
        "shadowColor": "rgba(34, 38, 36, 0.15)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 5,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignItems": "center",
            "color": "#222624",
            "flexWrap": "wrap",
            "fontFamily": "Helvetica Neue",
            "fontWeight": "bold",
            "justifyContent": "center",
            "position": "relative",
            "width": "100%",
          },
          Object {
            "fontSize": 35,
            "lineHeight": 39.97,
          },
          Object {},
        ]
      }
    >
      The order i190519-24047 is placed
    </Text>
    <Text
      style={
        Array [
          Object {
            "flexWrap": "wrap",
            "fontFamily": "Helvetica",
            "fontWeight": "400",
          },
          Object {
            "color": "#222624",
            "fontSize": 16,
            "lineHeight": 24,
          },
          Object {
            "marginTop": 16,
          },
        ]
      }
    >
      We will check the availability of colors and delivery details and send you a payment form. This usually takes no more than 5 minutes.
    </Text>
  </View>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#399AE5",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 56,
        "justifyContent": "center",
        "minHeight": 36,
        "minWidth": "100%",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#399AE5",
            "fontFamily": "Helvetica Neue",
            "fontWeight": "500",
            "textAlign": "center",
          },
          Object {
            "color": "#399AE5",
            "fontSize": 16,
          },
        ]
      }
    >
      Go to catalogue
    </Text>
  </View>
</View>
`;
